README.rst

```diff
@@ -17,18 +17,151 @@ Requirements: Python >= 3.9.
+    # development, since building PDF and ePUB takes time. It also supports
+    # the VERSION argument.
+    make build_html
+
+.. note::
+
+    Please refer the "`build options" <#build-options>`_section of this
+    configuration for a complete reference of the available options.
 
 5. Open the generated HTML files in your browser.
 
+Build options
+-------------
+
+Building different formats
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the ``make build`` command will generate documentation in all
+supported output formats: HTML, PDF, and ePUB.
+
+If you want to generate the documentation in specific formats,
+you can specify the ``FORMATS`` argument. The ``FORMATS`` argument
+accepts a comma-separated list of formats to generate.
+The supported formats are:
+
+- ``html``: HTML documentation
+- ``pdf``: PDF documentation
+- ``epub``: ePUB documentation
+
+For example, to generate only HTML and PDF documentation, you can run:
+
+.. code-block:: shell
+
+    make build FORMATS=html,pdf
+
+To generate all supported formats, just omit the ``FORMATS`` argument:
+
+.. code-block:: shell
+
+    make build
+
+.. code-block:: shell
+
+    # This command will only generate HTML
+    make build FORMATS=html
+
+Building specific version
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the ``make build`` command will generate documentation for
+all the versions defined in ``config.yml``.
+
+If you want to generate the documentation for a specific version,
+you can do so by using the ``VERSION`` argument. ``VERSION`` accepts any
+version that is specified in the ``config.yml`` file.
+
+For example, if you want to generate documentation for the
+``dev`` version, you can run:
+
+.. code-block:: shell
+
+    make build VERSION=dev
+
+This is useful if you only want to generate documentation for the
+version you are currently working on, or if you want to generate
+documentation for a specific version without having to rebuild all
+the other versions as well.
+
+Overriding a module of a version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``make build`` command is programmed to generate the documentation
+for the modules that are defined in the ``config.yml`` file. Sometimes,
+it may be necessary to override the branch/remote of a module defined in the
+``config.yml`` file to build the documentation for a specific version
+or to test a specific commit/branch of a module.
+
+You can do so by using the ``MODULES`` argument. ``MODULES`` accepts a
+comma separated string where each item is of the following format:
+
+.. code-block:: text
+
+    version=<openwisp-version>,repository=<repo-owner>/<repo-name>,branch=<branch-name>
+
+E.g. if you want to build the documentation for the ``dev`` version, but want
+to use the ``feature`` branch of openwisp-controller of your fork, then
+the command will be:
+
+.. code-block:: shell
+
+    make build MODULES="version=dev:repository=<your-username>/openwisp-controller:branch=feature"
```

Comment:
If we already have the version flag, why allow to specify the version here too?

Seems redundant to me. 

Why not just:

```
make build VERSION=dev MODULES="repository=<your-username>/openwisp-controller:branch=feature"
```

Can you allow something like this?

```
make build VERSION=dev MODULES="openwisp-users,openwisp-controller"
```

In this case, if nothing is specified, we can assume this is the list of module we want to build, so use the defaults provided in the config YAML but build only these modules instead of all of them.

Reply:
> Can you allow something like this?
> 
> ```
> make build VERSION=dev MODULES="openwisp-users,openwisp-controller"
> ```
> 
> In this case, if nothing is specified, we can assume this is the list of module we want to build, so use the defaults provided in the config YAML but build only these modules instead of all of them.

How will this help us in using the script programmatically?

If you want to do this for quick builds, you can temporarily comment out the modules from config.yml

Reply:
> If we already have the version flag, why allow to specify the version here too?
> 
> Seems redundant to me.
> 
> Why not just:
> 
> ```
> make build VERSION=dev MODULES="repository=<your-username>/openwisp-controller:branch=feature"
> ```

The `VERSION` argument determines which version to build. The first `version` key in the `MODULES` determine which version to add this module. 

E.g. you can have something like this 

```
make build MODULES="version=dev:repository=<your-username>/openwisp-controller:branch=feature,version=22.05:repository=<your-username>/openwisp-controller:branch=22.05"
```

Reply:
Ok let's leave this for now.